crates/red_knot_project/src/metadata.rs

```diff
@@ -321,7 +321,7 @@ mod tests {
 
         system
             .memory_file_system()
-            .write_files([(root.join("foo.py"), ""), (root.join("bar.py"), "")])
+            .write_files_all([(root.join("foo.py"), ""), (root.join("bar.py"), "")])
```

Comment:
I'm not very happy with this name. The `MemoryFileSystem::write_files` and `write_file` methods used to create any missing directories. This is inconsistent with what other systems do but is very convenient in tests. 

That's why I renamed the existing `write_files` and `write_file` methods to `write_files_all` and `write_file_all` (in resemblence of `create_dir_all`) and created a new `write_file` method that errors when the parent directory doesn't exist. However, I don't think it's a very good name.